src/timeline/TimelineModel.h

```diff
@@ -218,7 +218,9 @@ class TimelineModel : public QAbstractListModel
         Q_INVOKABLE void redactEvent(QString id);
         Q_INVOKABLE int idToIndex(QString id) const;
         Q_INVOKABLE QString indexToId(int index) const;
-        Q_INVOKABLE void cacheMedia(QString eventId);
+        Q_INVOKABLE void openMedia(QString eventId);
+        Q_INVOKABLE void cacheMedia(QString eventId,
+                                    std::function<void(const QString filename)> callback = { });
```

Comment:
This fails to compile on some platforms. Can you overload these, i.e. have one cacheMedia, that only takes a QString and the other one with to args? You can still let one call the other, but make only the one with just one arg exposed to Qml (with the Q_INVOKABLE).

Reply:
Is this ok?

Reply:
Seems to compile now